File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
  // Prompt user for an amount of money
  float amount;
  do
  {
    amount = get_float("Change owed: $");
  } while (amount < 0);

  // Convert amount to cents
  int cents = round(amount * 100);

  // Use the largest coins possible, keeping track of the number of coins used
  int coins = 0;

  // Use 0.25$
  while (cents >= 25)
  {
    cents -= 25;
    coins++;
  }

  // Use 0.10$
  while (cents >= 10)
  {
    cents -= 10;
    coins++;
  }

  // Use 0.05$
  while (cents >= 5)
  {
    cents -= 5;
    coins++;
  }

  // Use 0.01$
  while (cents >= 1)
  {
    cents -= 1;
    coins++;
  }

  // Print the number of coins used
  printf("%i\n", coins);
}

File: pset6/cash/cash.py
def main():
  cash = get_positive_float("Change owed: $")
  coins = get_coins(cash)
  print(coins)
  pass


def get_float(arg: str = "") -> float:
  while True:
    try:
      return float(input(arg))
    except:
      continue


def get_positive_float(arg: str = "", max: float = None) -> float:
  while True:
    x: float = get_float(arg)
    if max and x > max:
      continue
    if x > 0:
      return x


def get_coins(cash: float) -> int:
  cash = cash * 100
  coins: int = 0
  while cash >= 25:
    cash -= 25
    coins += 1
  while cash >= 10:
    cash -= 10
    coins += 1
  while cash >= 5:
    cash -= 5
    coins += 1
  while cash >= 1:
    cash -= 1
    coins += 1
  return coins


main()
